syntax = "proto3";

package payment_service;

service PaymentService {
    rpc CreatePayment(CreatePaymentRequest) returns (PaymentResponse);
    rpc GetPayment(GetPaymentRequest) returns (PaymentResponse);
}

message CreatePaymentRequest {
    string order_id = 1;
    double amount = 2;
}

message PaymentResponse {
    string payment_id = 1;
    string order_id = 2;
    double amount = 3;
    string status = 4;
}

message GetPaymentRequest {
    string payment_id = 1;
}